File: Problems/Combinations.java
class Solution {
    public List<List<Integer>> combine(int n, int k) {
        List<List<Integer>> result = new LinkedList<>();
        if (n < k || k == 0) return result;
        
        result = combine(n - 1, k - 1);
        if (result.isEmpty()) {
            result.add(new LinkedList<Integer>());
        }
        for (List<Integer> list : result) {
            list.add(n);
        }
        result.addAll(combine(n - 1, k));
        return result;
    }
}

File: AlgorithmStudyPlan/Combinations.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        if k == 1:
            return [[i] for i in range(1, n+1)]
        
        else:
            previousValues = self.combine(n,k-1)
            newValues = []
            for i in previousValues:
                for j in range(i[-1] + 1, n + 1):
                    newValues.append(i + [j])
            return newValues
